[WIP] Addressing issues for Make chapter

GjjvdBurg opened the pull request:
**This is a work in progress, please don't merge yet.**

This PR will address all todo items in alan-turing-institute/the-turing-way#409

GjjvdBurg committed:
use sudo for pacman

GjjvdBurg committed:
Add additional reasons for using make

GjjvdBurg committed:
Add remark on tabs vs spaces

GjjvdBurg committed:
Minor language fixes

GjjvdBurg committed:
test fix syntax highlighting

GjjvdBurg committed:
fix makefile syntax highlighting everywhere

GjjvdBurg committed:
Reorder reasons for using make

GjjvdBurg committed:
Use tabs instead of spaces

GjjvdBurg committed:
add link to installation instructions

GjjvdBurg committed:
Fix link to section

GjjvdBurg committed:
fix indentation in make code

GjjvdBurg committed:
Initial glossary

GjjvdBurg committed:
Writing improvements

GjjvdBurg committed:
update table of contents

GjjvdBurg committed:
remove unnecessary heading depth

GjjvdBurg committed:
add shell tutorial link

GjjvdBurg committed:
move source repo to ATI organisation

GjjvdBurg committed:
Organise the further reading section

I think this is more appropriate than a
Bibliography, as I didn't use anything other
than the make reference manual while writing.

GjjvdBurg committed:
update toc

GjjvdBurg committed:
formatting

GjjvdBurg committed:
Add paragraph on Snakemake

GjjvdBurg committed:
Make preface to alternatives more neutral

GjjvdBurg committed:
add schematic for research project

GjjvdBurg committed:
shrink figure a bit

GjjvdBurg committed:
move figure

GjjvdBurg committed:
add caption to make graph

GjjvdBurg committed:
Add more details to setting up section

GjjvdBurg committed:
Make sure Make is capitalised when appropriate

GjjvdBurg committed:
center captions

GjjvdBurg committed:
add note on dollar sign

GjjvdBurg committed:
minor text updates

rainsworth commented:
Hi @GjjvdBurg - just a heads up that we've been merging the book back into the main repo: https://github.com/alan-turing-institute/the-turing-way

As such, would it be possible for you to make your pull request to that repo instead? I apologise profusely for the inconvenience, thank you so much for bearing with us while we iron these aspects of working on an open project!

Please let me know if you have any questions about this or need any help.

GjjvdBurg commented:
@rainsworth Not a problem Rachael! Once I've finished everything here, I'll just recreate the PR on the other repo.

GjjvdBurg committed:
Updates to big_data section

GjjvdBurg committed:
Add an image showing the result

GjjvdBurg committed:
Add a section on debugging makefiles

GjjvdBurg committed:
Minor spelling and writing changes

GjjvdBurg committed:
Add an advanced section on canned recipes

This is taken from the original tutorial I wrote.
I think it's an advanced feature of Make that some
people might find useful, but I don't want to add
it to the main tutorial section to keep that concise.
